### PR TITLE
Add weather api #131

### PR DESCRIPTION
**Pull request by daltonlim**
GitHub Issue (If applicable): #131 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the new behavior?
Retrieves weather data from auckland.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [x] Added new tests to the test suite for the feature
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Checked licensing of Frameworks/Libraries (if applicable) 

## Other information

Internal Issue (If applicable):

Pair programmed with @res550 @tigerj2 @Ichu618. Added api calls for weather

### CODE REVIEW

File: server/src/main/java/com/feeder/server/provider/weather/WeatherDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  private WebClient client = WebClient.create();
+
+  @Override
+  public Flux<WeatherData> getFeed() {
+    // Gets all "Ask Weather" posts. Build a WeatherData type for each forecast retrieved
+    return client
+        .get()
+        .uri(
+            "https://api.openweathermap.org/data/2.5/weather?q=auckland&appid=fed5f6b010d51f1e212951f886b8d87f")
```
Comment (lucycy98) [inline server/src/main/java/com/feeder/server/provider/weather/WeatherDataProvider.java:20]: should the API key be in the applications.properties file?

File: server/src/main/java/com/feeder/server/model/WeatherData.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  public static Builder newBuilder() {
+    return new AutoValue_WeatherData.Builder();
+  }
+
+  @Override
+  public Type feedType() {
+    return Type.WEATHER;
+  }
+
+  /** This property is required for serialization. This means all posts must contain an id. */
```
Comment (lucycy98) [inline server/src/main/java/com/feeder/server/model/WeatherData.java:23]: if it's required, then is there a reason id is declared as optional?

File: server/src/main/java/com/feeder/server/model/WeatherData.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  @JsonProperty("dt")
+  public abstract Optional<Object> dt();
+
+  @JsonProperty("clouds")
+  public abstract Optional<Object> clouds();
+
+  @JsonProperty("sys")
+  public abstract Optional<Object> sys();
+
+  @JsonProperty("id")
```
Comment (lucycy98) [inline server/src/main/java/com/feeder/server/model/WeatherData.java:54]: should this be a string?

File: server/README.md [hunk 4]
```diff
@@ -30,6 +30,6 @@ have any credentials see section [Creating new credentials](#creating-new-creden
 - **Reddit** - These attributes can be obtained after creating an app [here](https://www.reddit.com/prefs/apps). Choose a `script` app type. Use the username and password of the account that created the app. 
 - **Github** - [Basic authentication](https://developer.github.com/v3/auth/#basic-authentication) is used, so use the username and password of a newly created GitHub account.
 - **HackerNews** - Since HackerNews has a public API only, authentication is not required.
-
+- **Weather** - An api key must be retrieved from [OpenWeatherApp](https://openweathermap.org/api)
```
Comment (Kalashnikkov) [inline server/README.md:33]: Could you also add this into the main README.md as well!

File: server/src/main/java/com/feeder/server/model/GenericData.java [hunk 5]
```diff
@@ -10,7 +10,8 @@
     GITHUB,
     HACKERNEWS,
     TWITTER,
-    COVIDNINETEEN
+    COVIDNINETEEN,
```
Comment (Kalashnikkov) [inline server/src/main/java/com/feeder/server/model/GenericData.java:13]: Has this feature been implemented already?

File: server/src/main/java/com/feeder/server/model/WeatherData.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  public static Builder newBuilder() {
+    return new AutoValue_WeatherData.Builder();
+  }
+
+  @Override
+  public Type feedType() {
+    return Type.WEATHER;
+  }
+
+  @JsonProperty("name")
```
Comment (cwon880) [inline server/src/main/java/com/feeder/server/model/WeatherData.java:23]: Are all the properties listed below optional? So nothing is required for serialization?

File: server/src/main/java/com/feeder/server/model/GenericData.java (hunk 5 above)
Comment (daltonlim) [reply server/src/main/java/com/feeder/server/model/GenericData.java:13]: Yea, thats just a merge conflict fix

File: server/src/main/java/com/feeder/server/model/WeatherData.java (hunk 6 above)
Comment (daltonlim) [reply server/src/main/java/com/feeder/server/model/WeatherData.java:23]: Good catch, I'll update it now.

File: server/README.md (hunk 4 above)
Comment (daltonlim) [reply server/README.md:33]: Updates

**Pull request merged**